packages/core/src/common/command.ts

```diff
@@ -141,6 +149,11 @@ export interface CommandService {
      * It can be used to install or activate a command handler.
      */
     readonly onWillExecuteCommand: Event<WillExecuteCommandEvent>;
+
+    /**
+     * An event is emmited when a command has been successfully executed.
+     */
+    readonly onDidExecuteCommand: Event<DidExecuteCommandEvent>;
```

Comment:
To make it easier to match up "will execute" to "did execute" events, how about moving `onDidExecute` to the `WillExecuteCommandEvent`? So callers can do something like this:
```typescript
commandService.onWillExecuteCommand(({ commandId, onDidExecute }) => {
    if (commandId === MY_COMMAND) {
        showProgress();
        onDidExecute(hideProgress);
   }
});
```

Or equivalently pass a promise to the event for chaining.

Reply:
To pass a function that will be invoked when something completed as arg of the `onWill` event? Yep, it makes code shorter but in Theia code contribution guidelines they explicitly mention the `onWillX`, `onDidX` couple of events.

Reply:
IMO, it will be easier to merge upstream if we follow Theia's pattern as @adriano-repetti had it.